### PR TITLE
[COST-3796] Add AWS cost category settings API

### PR DESCRIPTION
**Pull request by myersCody**
## Jira Ticket

[COST-3796](https://issues.redhat.com/browse/COST-3796)

## Description

This change will add new api endpoints for manipulating and retrieving cost category keys. 

*Note*
- Migrations will be removed from this pr when they go into main. 

## Testing

1. load-test-customer-data
2. Hit the following api endpoint:
```
http://localhost:8000/api/cost-management/v1/settings/aws_category_keys/
```
Example:
```
"data": [
        {
            "uuid": "be9b46e1-ae73-425c-a3bc-738018d2a41f",
            "key": "Charge type",
            "enabled": true
        },
        {
            "uuid": "ec0de1d4-d96d-4323-8929-1af5ff80db69",
            "key": "CostCenter",
            "enabled": true
        },
        {
            "uuid": "8f7e988d-488e-43df-9b1f-ad8deca3983d",
            "key": "env",
            "enabled": true
        },
        {
            "uuid": "27288963-463d-40bd-9a7a-44365c006095",
            "key": "Organization",
            "enabled": true
        },
        {
            "uuid": "40288766-9ba0-4f18-8325-ad40e79f43b8",
            "key": "OUs",
            "enabled": true
        }
    ]
```
Grab one of the uuids from the return
3. Disable a category key:
URL: http://localhost:8000/api/cost-management/v1/settings/aws_category_keys/disable/
Type: PUT
Body: 
```
{
    "ids": ["8f7e988d-488e-43df-9b1f-ad8deca3983d"]
}
```
5. Check the logs to make sure the cache is invalidated:
```
koku_server         | [2023-06-14 20:35:29,035] INFO None Invalidated request cache for
koku_server         | 	tenant: org1234567
koku_server         | 	cache_key_prefix: aws-view
koku_server         | [2023-06-14 20:35:29,038] INFO None Invalidated request cache for
koku_server         | 	tenant: org1234567
koku_server         | 	cache_key_prefix: openshift-aws-view
koku_server         | [2023-06-14 20:35:29,041] INFO None Invalidated request cache for
koku_server         | 	tenant: org1234567
koku_server         | 	cache_key_prefix: openshift-all-view
```
6. Check the database:
```
postgres=# select * from reporting_awsenabledcategorykeys;
     key      | enabled |                 uuid                 
--------------+---------+--------------------------------------
 Charge type  | t       | 485a9bfe-c787-43bf-8417-def83eb15f0d
 Organization | t       | 7f8c5688-dc92-4b12-8d4f-c882faa65e67
 OUs          | t       | b6e8303b-6d58-49cc-9962-ab419e6b13c9
 CostCenter   | t       | 30bf5072-d2dd-4d07-9a4d-9d02a4c2048d
 env          | f       | 8f7e988d-488e-43df-9b1f-ad8deca3983d
```
7. Check resource types:
```
http://localhost:8000/api/cost-management/v1/resource-types/aws-categories/
```
This will check that the cache got invalidated.
```
{
            "key": "env",
            "values": [
                "ephemeral",
                "prod",
                "stage"
            ],
            "enabled": "False"
        },
```
8. Enable cost category 
URL: http://localhost:8000/api/cost-management/v1/settings/aws_category_keys/disable/
Type: PUT
Body: 
```
{
    "ids": ["8f7e988d-488e-43df-9b1f-ad8deca3983d"]
}
```
9. check the database & the resource_types endpoint again to see that it has been enabled. 
## Notes

...

### CODE REVIEW

**Review by lcouzens — CHANGES_REQUESTED** [review]
I'd like to better understand the need for using a UUID for enabling/disabling keys rather than just using the tables primary_key `key`?

File: koku/reporting/provider/aws/models.py [hunk 1]
```diff
@@ -283,6 +283,8 @@ class Meta:
         indexes = [models.Index(fields=["key", "enabled"], name="aws_enabled_category_key_index")]
         db_table = "reporting_awsenabledcategorykeys"
 
+    # TODO: remove uuid & migration since it will be handled in a separate PR.
```
[the remark this answers is not in the document]
Comment (myersCody) [reply koku/reporting/provider/aws/models.py:286]: This conversation is happening [here](https://github.com/project-koku/koku/pull/4464/files/ea9d962a081c4f308a7a2b33e83275f97f172e81#r1231529662). Essentially, Corey and I think it is a good idea to keep the way we enable/disable keys through the settings pages consistent. So, this change is to match the way we will be doing tag enablement in the future.

**Review by lcouzens — APPROVED** [review]

File: koku/reporting/provider/aws/models.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (lcouzens) [reply koku/reporting/provider/aws/models.py:286]: Migration changes look good, just make sure to merge the right PR first. :smirk:

**Review by lcouzens — APPROVED** [review]

**Pull request merged**